kratos/processes/simple_mortar_mapper_process.h

```diff
@@ -0,0 +1,474 @@
+                        
+                        Matrix var_origin_matrix;
+                        MortarUtilities::MatrixValue<TVarType, THist>(slave_geometry, mDestinyVariable, var_origin_matrix);
+        
+                        const Matrix var_destiny_matrix = (master_area/total_master_area) * prod(p_operator, var_origin_matrix);
+                        MortarUtilities::AddValue<TVarType, THist>(master_geometry, mDestinyVariable, var_destiny_matrix);
+                    }
+                }
+            }
+        }
+        
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * This method sets both variables (origin and destination) with the same variable
+     */
+    void SetVariable(TVarType ThisVariable)
+    {
+        mOriginVariable = ThisVariable;
+        mDestinyVariable = ThisVariable;
+    }
+    
+    /**
+     * This method sets both variables (origin and destination) in a separated way
+     */
+    void SetVariables(
+        TVarType OriginVariable,
+        TVarType DestinyVariable
+        )
+    {
+        mOriginVariable = OriginVariable;
+        mDestinyVariable = DestinyVariable;
+    }
+    
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "SimpleMortarMapperProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "SimpleMortarMapperProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+    ///@}
+    ///@name Member Variables
+    ///@{
+    
+    ModelPart& mrThisModelPart; // The model part to compute
+    TVarType mOriginVariable;   // The origin variable to map
+    TVarType mDestinyVariable;  // The destiny variable to map
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+    
+    /**
+     * This method computes the nodal area
+     */
+    void ComputeNodalArea()
+    {
+        NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+        const int num_nodes = static_cast<int>(nodes_array.size()); 
+        
+        // We set to zero
+        #pragma omp parallel for
+        for(int i = 0; i < num_nodes; i++) 
```

Comment:
`++i`
Doesn't make a difference here but I think it is good practice

Reply:
why?

Reply:
see here: https://stackoverflow.com/questions/7436247/concept-difference-between-pre-and-post-increment-operator-for-stl
@roigcarlo please correct me if this is outdated

Reply:
In general is a good practice but this particular case will be the same.

Since this topic appears a lot, consider the operators as if there were functions like these:

```C++
// Pre-inc
Type& operator ++ (Type& n) {
    n = n + 1;
    return n;
}

// Post-inc
Type operator ++ (Type& n)  {
    Type tmp = n;
    n = n + 1;
    return tmp;
}
```

The efficiency problem is in the copy (Type tmp = n). If it is correctly implemented like in built in types or iterators there is no problem. If your class is iterable or iterators are overloaded and the copy constructor is expensive, the performance impact will be huge.

If we were using OpenMP 4 and we rewrite the code like:

```C++
#pragma omp parallel for
for(auto it_node = nodes_array.begin(); it_node != nodes_array.end(); ++it_node ) {
    it_node->SetValue(NODAL_AREA, 0.0);
}
```

Then it would maybe be faster (probably not) because we avoid the implicit copy, but as @philbucher reference suggests it will never be slower.

Personaly I like to put ++X only when it is necessary as it indicates me that the copy is non-trivial, but I know that is not the most extended opinion so take this just as aditional info.